### PR TITLE
Add a tutorial for the OpenTelemetry Spring Starters

### PR DESCRIPTION
**Pull request by mabdinur**
Quick guide on how to set up and use the Open Telemetry Starters

- clean up other spring docs

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

**Comment by iNikem** [comment]
@mabdinur please rebase this PR to the current master branch to allow CircleCI to succeed.

File: instrumentation/spring/README.md [hunk 1]
```diff
@@ -592,7 +592,396 @@ To create a sample trace enter `localhost:8080/message` in a browser. This trace
...
+```
+
+#### Gradle
+```gradle
+implementation "io.opentelemetry.instrumentation:opentelemetry-spring-starter:OPENTELEMETRY_VERSION"
+```
+
+### Create two Spring Projects
+
+Using the [spring project initializer](https://start.spring.io/), we will create two spring projects.  Name one project `MainService` and the other `TimeService`. Make sure to select maven, Spring Boot 2.3, Java, and add the spring-web dependency. After downloading the two projects include the OpenTelemetry dependencies listed above.  and they will be dealing with time.
```
Comment (trask) [inline instrumentation/spring/README.md:623]:
```suggestion
Using the [spring project initializer](https://start.spring.io/), we will create two spring projects.  Name one project `MainService` and the other `TimeService`. Make sure to select maven, Spring Boot 2.3, Java, and add the spring-web dependency. After downloading the two projects include the OpenTelemetry dependencies listed above.
```

File: instrumentation/spring/README.md [hunk 2]
```diff
@@ -592,7 +592,396 @@ To create a sample trace enter `localhost:8080/message` in a browser. This trace
...
+## Set Tracer name
+opentelemetry.trace.tracer.name=time_service
+opentelemetry.trace.tracer.samplerProbability=1
+
+## Default configurations
+#opentelemetry.trace.web.enabled = True
+#opentelemetry.trace.httpclients.enabled = True
+#opentelemetry.trace.tracer.samplingProbablity=1
+#opentelemetry.trace.exporter.loggin.enabled=True
+#opentelemetry.trace.aspects.enabled=True
```
Comment (trask) [inline instrumentation/spring/README.md:689]:
```suggestion
#opentelemetry.trace.web.enabled=true
#opentelemetry.trace.httpclients.enabled=true
#opentelemetry.trace.tracer.samplingProbablity=1
#opentelemetry.trace.exporter.loggin.enabled=true
#opentelemetry.trace.aspects.enabled=true
```

File: instrumentation/spring/README.md [hunk 3]
```diff
@@ -592,7 +592,396 @@ To create a sample trace enter `localhost:8080/message` in a browser. This trace
...
+
+```
+
+Check out [OpenTelemetry Spring Boot AutoConfigure](spring-boot-autoconfigure/README.md) to learn more.
+
+
+
+### TimeService
+
+Configure the main class in your `Time Service` project to match the file below. Here we use the Tracer bean provided by the OpenTelemetry to create an internal span and set some additional events and attributes.
```
Comment (trask) [inline instrumentation/spring/README.md:699]:
```suggestion
Configure the main class in your `Time Service` project to match the file below. Here we use the Tracer bean provided by the OpenTelemetry starter to create an internal span and set some additional events and attributes. 
```

File: instrumentation/spring/README.md [hunk 4]
```diff
@@ -592,7 +592,396 @@ To create a sample trace enter `localhost:8080/message` in a browser. This trace
...
+     @WithSpan(kind=Span.Kind.SERVER)
+     public void withSpanMethod() {}
+  }
+}
+```
+
+
+### Generating Trace - LoggingSpanExporter
+
+To generate a trace run MainServiceApplication and TimeServiceApplication and then send a request to `localhost:8080/message`. Shown below is the output of the the default span exporter - (LoggingSpanExporter)[https://github.com/open-telemetry/opentelemetry-java/tree/master/exporters/logging].
```
Comment (trask) [inline instrumentation/spring/README.md:755]: is LoggingSpanExporter the default, or does user need to add dependency for it too?
```suggestion
To generate a trace, run MainServiceApplication and TimeServiceApplication, and then send a request to `localhost:8080/message`. Shown below is the output of the default span exporter - (LoggingSpanExporter)[https://github.com/open-telemetry/opentelemetry-java/tree/master/exporters/logging].
```

File: instrumentation/spring/README.md [hunk 5]
```diff
@@ -592,7 +592,396 @@ To create a sample trace enter `localhost:8080/message` in a browser. This trace
...
+http.client_ip=AttributeValueString{stringValue=127.0.0.1}}, totalAttributeCount=8, totalRecordedEvents=0, 
+status=Status{canonicalCode=OK, description=null}, name=WebMVCTracingFilter.doFilterInteral, endEpochNanos=1598409410921631068, hasEnded=true
+}
+
+```
+
+
+### Exporter Starters 
+
+To configure the OpenTelemetry tracing with the OTLP, Zipkin, or Jaeger span exporters replace the OpenTelemtry Spring Starter dependency with one of the artifacts listed below:
```
Comment (trask) [inline instrumentation/spring/README.md:823]:
```suggestion
To configure OpenTelemetry tracing with the OTLP, Zipkin, or Jaeger span exporters replace the OpenTelemetry Spring Starter dependency with one of the artifacts listed below:
```

File: instrumentation/spring/README.md [hunk 6]
```diff
@@ -592,7 +592,396 @@ To create a sample trace enter `localhost:8080/message` in a browser. This trace
...
+#opentelemetry.trace.exporters.jaeger.spantimeout=1s
+#opentelemetry.trace.exporters.zipkin.servicename=unknown
+#opentelemetry.trace.exporters.zipkin.endpoint=http://localhost:9411/api/v2/spans
+```
+
+### Sample Trace Zipkin
+
+To generate a trace using the zipkin exporter follow the steps below: 
+ 1. Replace `opentelemetry-spring-starter` with `opentelemetry-zipkin-starter` in your pom or gradle build file
+ 2. Use the Zipkin [quick starter](https://zipkin.io/pages/quickstart) to download and run the zipkin exporter jar
```
Comment (trask) [inline instrumentation/spring/README.md:887]:
```suggestion
 2. Use the Zipkin [quick starter](https://zipkin.io/pages/quickstart) to download and run the zipkin executable jar 
```

File: instrumentation/spring/README.md [hunk 7]
```diff
@@ -592,7 +592,396 @@ To create a sample trace enter `localhost:8080/message` in a browser. This trace
...
+```
+
+### Sample Trace Zipkin
+
+To generate a trace using the zipkin exporter follow the steps below: 
+ 1. Replace `opentelemetry-spring-starter` with `opentelemetry-zipkin-starter` in your pom or gradle build file
+ 2. Use the Zipkin [quick starter](https://zipkin.io/pages/quickstart) to download and run the zipkin exporter jar 
+    - Ensure the zipkin endpoint matches the default value listed in your application properties
+ 3. Run `MainServiceApplication.java` and `TimeServiceApplication.java`
+ 4. Use your favorite browser to send a request to `localhost:8080/message`
```
Comment (trask) [inline instrumentation/spring/README.md:890]:
```suggestion
 4. Use your favorite browser to send a request to `http://localhost:8080/message`
```

File: instrumentation/spring/README.md [hunk 8]
```diff
@@ -592,7 +592,396 @@ To create a sample trace enter `localhost:8080/message` in a browser. This trace
...
+ 5. Navigate to `localhost:9411` to see your trace
```
Comment (trask) [inline instrumentation/spring/README.md:891]:
```suggestion
 5. Navigate to `http://localhost:9411` to see your trace
```

File: instrumentation/spring/README.md (hunk 4 above)
Comment (mabdinur) [reply instrumentation/spring/README.md]: LoggingSpanExporter is the default exporter provided by the opentelemetry-spring-starter. Users won't need to add this dependency.

**Pull request merged**